backend/src/graphql/resolver/UserResolver.test.ts

```diff
@@ -1578,39 +1693,83 @@ describe('UserResolver', () => {
+                  variables: { userId: user.id, role: ROLE_NAMES.ROLE_NAME_ADMIN },
+                })
+                expect(result).toEqual(
+                  expect.objectContaining({
+                    data: {
+                      setUserRole: ROLE_NAMES.ROLE_NAME_ADMIN,
+                    },
+                  }),
+                )
+              })
+
+              it('stores the ADMIN_USER_ROLE_SET event in the database', async () => {
+                const userContact = await UserContact.findOneOrFail({
+                  where: { email: 'bibi@bloxberg.de' },
+                  relations: ['user'],
+                })
+                const adminContact = await UserContact.findOneOrFail({
+                  where: { email: 'peter@lustig.de' },
+                  relations: ['user'],
+                })
+                await expect(DbEvent.find()).resolves.toContainEqual(
+                  expect.objectContaining({
+                    type: EventType.ADMIN_USER_ROLE_SET,
+                    affectedUserId: userContact.user.id,
+                    actingUserId: adminContact.user.id,
+                  }),
+                )
+              })
+            })
+
+            describe('to moderator', () => {
               it('returns date string', async () => {
                 const result = await mutate({
                   mutation: setUserRole,
-                  variables: { userId: user.id, isAdmin: true },
+                  variables: { userId: user.id, role: ROLE_NAMES.ROLE_NAME_MODERATOR },
                 })
                 expect(result).toEqual(
                   expect.objectContaining({
                     data: {
-                      setUserRole: expect.any(String),
+                      setUserRole: ROLE_NAMES.ROLE_NAME_MODERATOR,
                     },
                   }),
                 )
                 expect(new Date(result.data.setUserRole)).toEqual(expect.any(Date))
               })
 
               it('stores the ADMIN_USER_ROLE_SET event in the database', async () => {
-                const userConatct = await UserContact.findOneOrFail({
+                const userContact = await UserContact.findOneOrFail({
                   where: { email: 'bibi@bloxberg.de' },
                   relations: ['user'],
                 })
-                const adminConatct = await UserContact.findOneOrFail({
+                const adminContact = await UserContact.findOneOrFail({
                   where: { email: 'peter@lustig.de' },
                   relations: ['user'],
                 })
                 await expect(DbEvent.find()).resolves.toContainEqual(
                   expect.objectContaining({
                     type: EventType.ADMIN_USER_ROLE_SET,
-                    affectedUserId: userConatct.user.id,
-                    actingUserId: adminConatct.user.id,
+                    affectedUserId: userContact.user.id,
+                    actingUserId: adminContact.user.id,
```

Comment:
i will not agree, because the test compares on the previous loaded userContact.user.id and not on the prepared user and its id.
For sure the logic looks simpler, but it is not the requested comparison even if the user.id or admin.id will have the same value, the test fokus on the usercontacts userid.

Reply:
```
    await EVENT_ADMIN_USER_ROLE_SET(user, moderator)
````
This is how the event is saved. User and moderator do not use any user contact ids. I really do not get your point.

Reply:
see comment of line 1720-1728